SU2_CFD/src/driver_structure.cpp

```diff
@@ -3642,6 +3886,244 @@ void CFluidDriver::SetInitialMesh() {
   //}
 }
 
+CTurbomachineryDriver::CTurbomachineryDriver(char* confFile,
+    unsigned short val_nZone,
+    unsigned short val_nDim, SU2_Comm MPICommunicator) : CFluidDriver(confFile,
+        val_nZone,
+        val_nDim,
+        MPICommunicator) { }
+
+CTurbomachineryDriver::~CTurbomachineryDriver(void) { }
+
+void CTurbomachineryDriver::Run() {
+
+
+//  unsigned long ExtIter = config_container[ZONE_0]->GetExtIter();
+
+  int rank = MASTER_NODE;
+
+#ifdef HAVE_MPI
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+#endif
+
+
+  /*--- Run a single iteration of a multi-zone problem by looping over all
+   zones and executing the iterations. Note that data transers between zones
+   and other intermediate procedures may be required. ---*/
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    iteration_container[iZone]->Preprocess(output, integration_container, geometry_container,
+                                           solver_container, numerics_container, config_container,
+                                           surface_movement, grid_movement, FFDBox, iZone);
+  }
+
+  /* --- Update the mixing-plane interface ---*/
+  for (iZone = 0; iZone < nZone; iZone++) {
+    if(mixingplane)SetMixingPlane(iZone);
+  }
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    iteration_container[iZone]->Iterate(output, integration_container, geometry_container,
+                                        solver_container, numerics_container, config_container,
+                                        surface_movement, grid_movement, FFDBox, iZone);
+  }
+
+  for (iZone = 0; iZone < nZone; iZone++) {
+    iteration_container[iZone]->Postprocess(config_container, geometry_container,
+                                            solver_container, iZone);
+  }
+
+  if (rank == MASTER_NODE){
+    SetTurboPerformance(ZONE_0);
+  }
+
+
+}
+
+void CTurbomachineryDriver::SetMixingPlane(unsigned short donorZone){
```

Comment:
Yeah I see your point. However the new turbo driver is a child of the new FluidDriver, and only few routines are overwritten:
- the Run() routine because i need to use the iterator container in a slightly different way 
- the Monitor() routine in which I implemented ad hoc methods for ramping up the outlet pressure and the rotational speed to facilitate the convergence of multi-stage simulations. 

Indeed I could have implemented on the CFluidDriver, but there are two reason why I did not do it:
1. I wanted to avoid many if statements to customize the driver
2. (and more important) a clear structure of the turbo Driver also facilitate the implementation of his adjoint counterpart.

I' m pretty sure that when the structure is there we can always improve the code with future pull requests,  as we have been doing so far.  Anyway thanks for your valuable comments Giulio.

regards

Salvo